source/generators/ECECGenerator.cc

```diff
@@ -0,0 +1,158 @@
+// ----------------------------------------------------------------------------
+// nexus | ECECGenerator.cc
+//
+// This class is the primary generator for events consisting in the decay
+// of a double electron capture. The user must specify via configuration
+// parameters the atomic number, mass number and energy level of the
+// daughter isotope of interest.
+//
+// The NEXT Collaboration
+// ----------------------------------------------------------------------------
+
+
+#include "ECECGenerator.h"
+#include "GeometryBase.h"
+#include "DetectorConstruction.h"
+#include "FactoryBase.h"
+
+#include <G4GenericMessenger.hh>
+#include <G4RunManager.hh>
+#include <G4Event.hh>
+#include <G4AtomicShell.hh>
+#include <G4AtomicShellEnumerator.hh>
+#include <G4ParticleDefinition.hh>
+#include <G4PrimaryVertex.hh>
+#include <G4PrimaryParticle.hh>
+#include <G4DynamicParticle.hh>
+#include <G4UAtomicDeexcitation.hh>
+
+using namespace nexus;
+
+REGISTER_CLASS(ECECGenerator, G4VPrimaryGenerator)
+
+ECECGenerator::ECECGenerator():
+  G4VPrimaryGenerator(),
+  atomic_number_(-1),
+  shell_name_("K"), // K, L1, L2, L3, M1, ..., M5
+  region_("ACTIVE"),
+  msg_ (nullptr),
+
+  geom_ (nullptr),
+  shell_(nullptr),
+  atom_ (nullptr)
+{
+  msg_ = new G4GenericMessenger(this, "/Generator/ECEC/",
+                                "Control commands of the ECEC primary generator.");
+
+  G4GenericMessenger::Command& atomic_number_cmd =
+    msg_->DeclareProperty("atomic_number", atomic_number_,
+                          "Atomic number of the ion.");
+  atomic_number_cmd.SetParameterName("atomic_number", false);
+  atomic_number_cmd.SetRange("atomic_number > 0");
+
+  msg_->DeclareProperty("shell", shell_name_,
+                        "Shell from which the electron is captured.");
+
+  msg_->DeclareProperty("region", region_,
+                        "Region of the geometry where vertices will be generated.");
+
+}
+
+
+ECECGenerator::~ECECGenerator()
+{
+  delete msg_;
+  delete shell_;
+  delete atom_;
+}
+
+
+G4AtomicShellEnumerator ECECGenerator::GetShellID(G4String s)
```

Comment:
I get a compiler warning because of the variable name: `warning: declaration shadows a variable in namespace 'CLHEP'`. Could you use another name instead of `s`? Anything that is not related to units of measure.